src/main/java/org/folio/circulationbff/service/impl/CheckInServiceImpl.java

```diff
@@ -32,21 +35,114 @@
 public class CheckInServiceImpl implements CheckInService {
 
   private final CheckInClient checkInClient;
+  private final CirculationItemClient circulationItemClient;
+  private final SettingsService settingsService;
+  private final UserTenantsService userTenantsService;
   private final SearchService searchService;
   private final InventoryService inventoryService;
   private final SystemUserScopedExecutionService executionService;
   private static final String DCB_INSTANCE_ID = "9d1b77e4-f02e-4b7f-b296-3f2042ddac54";
 
   @Override
   public CheckInResponse checkIn(CheckInRequest request) {
-    log.info("checkIn: checking in item with barcode {} on service point {}",
-      request.getItemBarcode(), request.getServicePointId());
-    var response = checkInClient.checkIn(request);
+
+    String tenantId = determineTenantForCheckIn(request);
+
+    var response = tenantId == null
+      ? checkInLocally(request)
+      : checkInRemotely(request, tenantId);
+
     processCheckInResponse(response);
 
     return response;
   }
 
+  /**
+   * @return Tenant ID to proxy check in operation to; null means local check in.
+   */
+  private String determineTenantForCheckIn(CheckInRequest request) {
+    if (!settingsService.isEcsTlrFeatureEnabled() || !userTenantsService.isCentralTenant()) {
```

Comment:
This triggers a bug in UserTenantService#isCentralTenant.

Reply:
Using TenantService now